module/zfs/zfs_acl.c

```diff
@@ -2340,7 +2342,8 @@ zfs_has_access(znode_t *zp, cred_t *cr)
 	if (zfs_zaccess_aces_check(zp, &have, B_TRUE, cr) != 0) {
 		uid_t owner;
 
-		owner = zfs_fuid_map_id(ZTOZSB(zp), zp->z_uid, cr, ZFS_OWNER);
+		owner = zfs_fuid_map_id(ZTOZSB(zp), zfs_uid_read(ZTOI(zp)),
+					cr, ZFS_OWNER);
```

Comment:
SunOS style in to indent 4 spaces for continued lines, which differs from the kernel style.